notion_client/helpers.py

```diff
@@ -124,3 +134,169 @@ def is_equation_rich_text_item_response(rich_text: Dict[Any, Any]) -> bool:
+
+    trimmed = url_or_id.strip()
+
+    # Check if it's already a properly formatted UUID
+    uuid_pattern = re.compile(
+        r"^[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}$", re.IGNORECASE
+    )
+    if uuid_pattern.match(trimmed):
+        return trimmed.lower()
+
+    # Check if it's a compact UUID (32 chars, no hyphens)
+    compact_uuid_pattern = re.compile(r"^[0-9a-f]{32}$", re.IGNORECASE)
+    if compact_uuid_pattern.match(trimmed):
+        return _format_uuid(trimmed.lower())
+
+    # Try to parse as URL
+    try:
+        parsed = urlparse(trimmed)
+
+        # Check if it's a valid Notion domain
+        valid_domains = ["notion.so", "www.notion.so", "notion.site"]
+        if parsed.netloc not in valid_domains:
+            return None
+
+        # First, try to extract ID from the path (prioritized)
+        path = parsed.path
+        if path:
+            # Look for 32-character hex strings in the path
+            path_id_matches = re.findall(r"[0-9a-f]{32}", path, re.IGNORECASE)
+            if path_id_matches:
+                # Take the last match (most likely to be the main ID)
+                return _format_uuid(path_id_matches[-1].lower())
+
+        # If no path ID found, try query parameters
+        query_params = parse_qs(parsed.query)
+        for param in ["p", "id"]:
+            if param in query_params and query_params[param]:
+                candidate = query_params[param][0].strip()
+                if compact_uuid_pattern.match(candidate):
+                    return _format_uuid(candidate.lower())
+                elif uuid_pattern.match(candidate):
+                    return candidate.lower()
+
+        # Try fragment (for block IDs)
+        if parsed.fragment:
+            fragment = parsed.fragment
+            # Remove "block-" prefix if present
+            if fragment.startswith("block-"):
+                fragment = fragment[6:]
+            if compact_uuid_pattern.match(fragment):
+                return _format_uuid(fragment.lower())
+
+        return None
+
+    except Exception:
+        return None
+
+
+def extract_database_id(url_or_id: str) -> Optional[str]:
+    """Extract a database ID from a Notion URL or validate if it's already a valid ID.
+
+    This is an alias for extract_notion_id for clarity when working with databases.
+
+    Args:
+        url_or_id: A Notion database URL or ID string
+
+    Returns:
+        The extracted UUID in standard format (with hyphens) or None if invalid
+    """
+    return extract_notion_id(url_or_id)
+
+
+def extract_page_id(url_or_id: str) -> Optional[str]:
+    """Extract a page ID from a Notion URL or validate if it's already a valid ID.
+
+    This is an alias for extract_notion_id for clarity when working with pages.
+
+    Args:
+        url_or_id: A Notion page URL or ID string
+
+    Returns:
+        The extracted UUID in standard format (with hyphens) or None if invalid
+    """
+    return extract_notion_id(url_or_id)
```

Comment:
I'm not sure if these are useful  but since they've been merged in notion-sdk-js, let's add them here as well. Let's just give the same argument names as in notion-sdk-js: `database_url` and `page_url`